scylla-cdc/src/reader.rs

```diff
@@ -0,0 +1,266 @@
+use std::cmp::{max, min};
+use std::error::Error;
+use std::time;
+use std::time::SystemTime;
+
+use chrono;
+use futures::stream::StreamExt;
+use scylla::frame::value::Timestamp;
+use scylla::{Session, SessionBuilder};
+use tokio::time::sleep;
+
+type StreamID = Vec<u8>;
+
+pub struct Reader {
+    stream_id_vec: Vec<StreamID>,
+    lower_timestamp: chrono::Duration,
+    window_size: chrono::Duration,
+    safety_interval: chrono::Duration,
+    upper_timestamp: Option<chrono::Duration>,
+    sleep_interval: time::Duration,
+}
+
+impl Reader {
+    pub fn new(
+        stream_ids: Vec<StreamID>,
+        start_timestamp: chrono::Duration,
+        window_size: chrono::Duration,
+        safety_interval: chrono::Duration,
+        sleep_interval: time::Duration,
+    ) -> Reader {
+        Reader {
+            stream_id_vec: stream_ids,
+            lower_timestamp: start_timestamp,
+            window_size,
+            safety_interval,
+            upper_timestamp: std::option::Option::None,
+            sleep_interval,
+        }
+    }
+
+    // Mutable access.
+    fn lower_timestamp_mut(&mut self) -> Option<chrono::Duration> {
+        self.upper_timestamp
+    }
+
+    pub fn set_lower_timestamp(&mut self, new_lower_timestamp: chrono::Duration) {
+        self.upper_timestamp = Some(new_lower_timestamp);
+    }
```

Comment:
Are those functions properly named? Because they mention lower_timestamp in their names but they deal with upper_timestamp.

Reply:
Those functions are not used anywhere. If they're part of the required interface could you add some tests for them?